amplipi/streams.py

```diff
@@ -795,10 +795,28 @@ def send_cmd(self, cmd):
     """
     try:
       if cmd in self.supported_cmds:
+        if cmd == "love":
+          src_config_folder = f'{utils.get_folder("config")}/srcs/v{self.vsrc}'
+          with open(f'{src_config_folder}/.config/pianobar/currentSong', 'r+', encoding='utf-8') as file:
+            data = file.read()
+            chunks = data.split(",,,")
+            if len(chunks) >= 5:
+              liked_state = chunks[4].strip()
+              if liked_state == "0":
+                chunks[4] = "1"
+              elif liked_state == "1":
+                chunks[4] = "0"
+            modified_data = ",,,".join(chunks)
+            file.seek(0)
+            file.truncate()
+            file.write(modified_data)
+            file.flush()
```

Comment:
you open the file `r+` then truncate and write the file. [that's probably not what you want](https://docs.python.org/3/library/functions.html#open).

Reply:
r+ opens the file for reading, with no truncation. Using w+ would truncate the file immediately before I get a chance to read, making it so I cannot edit based on what data is already present

I read the file, see what there is, and make the changes I want to a local tempfile, then truncate and write afterwards as a means of overwriting the original file with the contents of the tempfile,. An alternative would be opening as r and then later opening as w but I feel that's likely less efficient than simply seeking and truncating manually

Reply:
ahhh gotcha, fair enough.